### PR TITLE
Quick Python 3 port of pg_dump_splitsort

### PR DESCRIPTION
**Pull request by connorsherson**
This PR is a quick port of the split script to Python 3. Only lightly tested at this time, but given the scripts relative simplicity I suspect it's likely fine.

I have taken the slightly weird choice and opted to keep both the py2 and py3 versions, particularly because I haven't used the other scripts myself and have not ported them. I didn't want to put the repo into a weird place where you'd need both versions to use all the scripts. It's obviously not an ideal long-term state, but this at least gets the repo part-way to working on maintained Python versions. I'm interested in your thoughts on this choice.

I have taken the time to comment in this PR the changes that I did to get the script running in Python 3.

### CODE REVIEW

File: pgtricks/pg_dump_splitsort_py3.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    except ValueError:
+        return s
+
+
+def linecomp(l1, l2):
+    p1 = l1.split('\t', 1)
+    p2 = l2.split('\t', 1)
+    v1 = try_float(p1[0])
+    v2 = try_float(p2[0])
+    result = (v1 > v2) - (v1 < v2)
```
Comment (connorsherson) [inline pgtricks/pg_dump_splitsort_py3.py]: cmp was removed in Python 3, I opted for the quick trick from [here](https://docs.python.org/3.0/whatsnew/3.0.html#ordering-comparisons)

File: pgtricks/pg_dump_splitsort_py3.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                elif SEQUENCE_SET_RE.match(line):
+                    pass
+                elif 1 <= counter < 9999:
+                    counter = 9999
+                    output = new_output('%04d_epilogue.sql' % counter)
+                buf.append(line)
+                flush()
+        else:
+            if line == '\\.\n':
+                copy_lines.sort(key=functools.cmp_to_key(linecomp))
```
Comment (connorsherson) [inline pgtricks/pg_dump_splitsort_py3.py]: list.sort no longer takes cmp-style funcs, and instead takes a "key" function. I opted for using the cmp_to_key helper, as pointed out in the [docs](https://docs.python.org/3/library/stdtypes.html#list.sort)

File: pgtricks/pg_dump_splitsort_py3.py (hunk 1 above)
Comment (akaihola) [reply pgtricks/pg_dump_splitsort_py3.py:26]: These comparisons fail for `VARCHAR` and `TEXT` fields with content that sometimes looks like an int or a float.

I solved this in #7 by converting both values to floats in one go and keeping as strings if either fails.

**Comment by akaihola** [comment]
I extended and superseded this pull request by #8. Your [py3 fix commit](https://github.com/akaihola/pgtricks/pull/8/commits/410a610967c7ca02fe2d56960b5919457b4325f6) is part of it @connorsherson, thanks for the contribution! I've also invited you as a collaborator on the project so you can review that pull request if you will.

**Comment by connorsherson** [comment]
@akaihola Pardon my extremely late reply! Man this feel's like a time capsule.

I'm glad my little fix was of value to you. And I appreciate the invite to review your work. Forgive me for accidental silent non-acceptance.

Thank you for the offer of a contributor mention! If you find yourself touching this repo again; sure I'd appreciate it. A simple bullet point with "Connor Sherson" will suffice. 

I'm not in the position of using this tool anymore myself, but thank you for creating it. It was very helpful at the time. I'm glad that I could contribute back a little.

**Pull request closed without merging**